server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/DateHistogramValuesSourceBuilder.java

```diff
@@ -70,9 +70,9 @@
         }, Histogram.INTERVAL_FIELD, ObjectParser.ValueType.LONG);
         PARSER.declareField(DateHistogramValuesSourceBuilder::timeZone, p -> {
             if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
-                return DateTimeZone.forID(p.text());
+                return ZoneId.of(p.text());
```

Comment:
Related to the version checks, should we support short IDs?  If not we should probably make a note in the breaking changes.